plugins/modules/s3_metrics_configuration.py

```diff
@@ -0,0 +1,246 @@
+      - "A dictionary of one or more tags used when evaluating a metrics filter"
+    required: false
+    aliases: ['filter_tag']
+    type: dict
+  state:
+    description:
+      - "Create or delete metrics configuration"
+    default: present
+    choices: ['present', 'absent']
+    type: str
+extends_documentation_fragment:
+- amazon.aws.aws
+- amazon.aws.ec2
+
+'''
+
+EXAMPLES = r'''
+# Note: These examples do not set authentication details, see the AWS Guide for details.
+
+- name: Create a metrics configuration that enables metrics for an entire bucket
+  community.aws.s3_metrics_configuration:
+    bucket_name: my-bucket
+    id: EntireBucket
+    state: present
+
+- name: Put a metrics configuration that enables metrics for objects starting with a prefix
+  community.aws.s3_metrics_configuration:
+    bucket_name: my-bucket
+    id: Assets
+    filter_prefix: assets
+    state: present
+
+- name: Put a metrics configuration that enables metrics for objects with specific tag
+  community.aws.s3_metrics_configuration:
+    bucket_name: my-bucket
+    id: Assets
+    filter_tag:
+      kind: asset
+    state: present
+
+- name: Put a metrics configuration that enables metrics for objects that start with a particular prefix and have specific tags applied
+  community.aws.s3_metrics_configuration:
+    bucket_name: my-bucket
+    id: ImportantBlueDocuments
+    filter_prefix: documents
+    filter_tags:
+      priority: high
+      class: blue
+    state: present
+
+- name: Delete metrics configuration
+  community.aws.s3_metrics_configuration:
+    bucket_name: my-bucket
+    id: EntireBucket
+    state: absent
+
+'''
+
+try:
+    import boto3
+    import botocore
+    from botocore.exceptions import ClientError, BotoCoreError
+except ImportError:
+    pass  # Handled by AnsibleAWSModule
+
+from ansible_collections.amazon.aws.plugins.module_utils.core import AnsibleAWSModule
+from ansible_collections.amazon.aws.plugins.module_utils.core import is_boto3_error_code
+from ansible_collections.amazon.aws.plugins.module_utils.ec2 import AWSRetry
+from ansible_collections.amazon.aws.plugins.module_utils.ec2 import ansible_dict_to_boto3_tag_list
+from ansible_collections.amazon.aws.plugins.module_utils.ec2 import boto3_tag_list_to_ansible_dict
+
+
+def _create_metrics_configuration(mc_id, filter_prefix, filter_tags):
+    payload = {
+        'Id': mc_id
+    }
+    if len(filter_tags) == 1 and filter_prefix is None:
+        payload['Filter'] = {
+            'Tag': ansible_dict_to_boto3_tag_list(filter_tags)[0]
+        }
+    elif len(filter_tags) == 0 and filter_prefix is not None:
+        payload['Filter'] = {
+            'Prefix': filter_prefix
+        }
+    elif len(filter_tags) > 0:
+        payload['Filter'] = {
+            'And': {
+                'Tags': ansible_dict_to_boto3_tag_list(filter_tags)
+            }
+        }
+    if filter_prefix is not None:
+        payload['Filter']['And']['Prefix'] = filter_prefix
```

Comment:
This results in a KeyError when the previous `if` blocks falls though the `len(filter_tags) == 0 and filter_prefix is not None` condition because `payload['Filter']['And']` does not exist when making this assignment.

The test at: https://github.com/ansible-collections/community.aws/blob/e504e52dffac34d5262d621f174b2c5b95d7903d/tests/integration/targets/s3_metrics_configuration/tasks/main.yml#L104 triggers the failure.

Reply:
🤦 the indentation issue